R/exponential_decay_corr_func.R

```diff
@@ -0,0 +1,50 @@
+#' Generates the correlation matrix for an exponential
+#' decay correlation function.
+#'
+#' @param corr_function_params Sturcuted as follows : {dist.matrix, phi, l}.
+#'
+#' @return Correlation matrix.
+#'
+exponential_decay_corr_func <- function(
+    corr_function_params = list(
+      dist_matrix = NULL,
+      phi = NULL,
+      l = NULL
+    )) {
+  # error managing
+  stopifnot(
+    "corr_function_params : NOT STRUCTURED CORRECTLY!!!\n
+             List names should be : {dist_matrix, phi, l}" =
+      names(corr_function_params) == c("dist_matrix", "phi", "l")
+  )
+  # presets
+  dist_matrix <- corr_function_params$dist_matrix
+  phi <- corr_function_params$phi
+  l <- corr_function_params$l
+  stopifnot(
+    "corr_function_params : DOES NOT CONTAIN VALID VALUES!!!\n
+             List should be {matrix, int, int}" =
+      is.matrix(dist_matrix) & is.numeric(phi) & is.numeric(l)
+  )
+  stopifnot(
+    "corr_function_params : DIST.MATRIX NOT SQUARE!!!\n
+             Make sure dist_matrix is square" =
+      nrow(dist_matrix) == ncol(dist_matrix)
+  )
+
+  # presets
+  corr_matrix <- matrix(
+    data = 0,
+    nrow = nrow(dist_matrix),
+    ncol = ncol(dist_matrix)
+  )
+
+  for (i in seq_len(nrow(dist_matrix))) {
+    for (j in seq_len(ncol(dist_matrix))) {
+      d_ij <- dist_matrix[i, j]
+      corr_matrix[i, j] <- exp(-(d_ij / phi)^l)
+    }
+  }
+
+  return(corr_matrix)
```

Comment:
Looking at this, it seems this would just return NULL if the default values are used. What does this mean in terms of the downstream functionality of the forward simulation? I am assuming we don't want to pass around a NULL matrix. 

We don't necessarily have to have default values for the function, especially if we want to create a switch for using the spatial correlation or not

Reply:
For this function, I added a stop function that doesn't let the function run if the values in corr_function_params are still NULL.  Is that fine?